beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -262,9 +264,17 @@ func (s *Store) Head(ctx context.Context) ([]byte, error) {
 	defer span.End()
 
 	head := s.JustifiedCheckpt().Root
-	filteredBlocks, err := s.getFilterBlockTree(ctx)
-	if err != nil {
-		return nil, err
+	filteredBlocks := make(map[[32]byte]*ethpb.BeaconBlock)
+	var err error
+	if featureconfig.Get().EnableBlockTreeCache {
+		s.filteredBlockTreeLock.Lock()
```

Comment:
RLock? Seems like you could process many readers in parallel with attestations coming in pubsub quickly